services/wallet/bridge/transfer.go

```diff
@@ -40,29 +40,37 @@ func (s *TransferBridge) CalculateFees(from, to *params.Network, token *token.To
 	return big.NewInt(0), big.NewInt(0), nil
 }
 
-func (s *TransferBridge) EstimateGas(fromNetwork *params.Network, toNetwork *params.Network, from common.Address, to common.Address, token *token.Token, toToken *token.Token, amountIn *big.Int) (uint64, error) {
-	estimation := uint64(0)
-	var err error
+func (s *TransferBridge) PackTxInputData(fromNetwork *params.Network, toNetwork *params.Network, from common.Address, to common.Address, token *token.Token, amountIn *big.Int) ([]byte, error) {
 	if token.Symbol == "ETH" {
-		estimation, err = s.transactor.EstimateGas(fromNetwork, from, to, amountIn, []byte("eth_sendRawTransaction"))
-		if err != nil {
-			return 0, err
-		}
+		return []byte("eth_sendRawTransaction"), nil
 	} else {
-		ethClient, err := s.rpcClient.EthClient(fromNetwork.ChainID)
-		if err != nil {
-			return 0, err
-		}
-
 		abi, err := abi.JSON(strings.NewReader(ierc20.IERC20ABI))
 		if err != nil {
-			return 0, err
+			return []byte{}, err
 		}
-		input, err := abi.Pack("transfer",
+		return abi.Pack("transfer",
 			to,
 			amountIn,
 		)
+	}
+}
 
+func (s *TransferBridge) EstimateGas(fromNetwork *params.Network, toNetwork *params.Network, from common.Address, to common.Address, token *token.Token, toToken *token.Token, amountIn *big.Int) (uint64, error) {
```

Comment:
maybe a todo: but it would be good to separate erc20 transfer from eth transfer.